### ISSUE TITLE
refactor: contract integration test should streamline the error for better testing

### ISSUE THREAD

**Issue by chungquantin** [r0gue-io/pop-node#119]
cc: @Daanvdplas 

```rs
fn decoded<T: Decode>(result: ExecReturnValue) -> T {
	match <T>::decode(&mut &result.data[2..]) {
		Ok(value) => value,
		Err(_) => panic!("\nTest failed by trying to decode `{:?}` into `T`\n", result),
	}
}
```

At the moment, any method failed will end up at line `13:39` which is inside the `decoded` method. This is very hard to debug. My recommendation is to refractor the `decoded` method to return `Result<T>` so it can streamline the error back to the main function. Something like:

```rs
fn decoded<T: Decode>(result: ExecReturnValue) -> Result<T> {
	<T>::decode(&mut &result.data[2..]).map_err("\nTest failed by trying to decode `{:?}` into `T`\n", result)
}
```

**Comment by Daanvdplas** [comment]
Yes this was something that could be improved, great suggestion!

**Comment by Daanvdplas** [comment]
Fixed by #130
